wallet/src/de/schildbach/wallet/ui/dashpay/CreateIdentityService.kt

```diff
@@ -0,0 +1,329 @@
+import org.bitcoinj.wallet.DeterministicSeed
+import org.bitcoinj.wallet.Wallet
+import org.bouncycastle.crypto.params.KeyParameter
+import org.dashevo.dashpay.BlockchainIdentity
+import org.dashevo.dpp.identity.Identity
+import org.dashevo.platform.Names
+import org.slf4j.LoggerFactory
+import java.io.IOException
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+
+class CreateIdentityService : LifecycleService() {
+
+    companion object {
+        private val log = LoggerFactory.getLogger(CreateIdentityService::class.java)
+
+        private const val ACTION_CREATE_IDENTITY = "org.dash.dashpay.action.CREATE_IDENTITY"
+
+        private const val EXTRA_USERNAME = "org.dash.dashpay.extra.USERNAME"
+
+        @JvmStatic
+        fun createIntent(context: Context, username: String): Intent {
+            return Intent(context, CreateIdentityService::class.java).apply {
+                action = ACTION_CREATE_IDENTITY
+                putExtra(EXTRA_USERNAME, username)
+            }
+        }
+    }
+
+    private val walletApplication by lazy { application as WalletApplication }
+    private val platformRepo by lazy { PlatformRepo(walletApplication) }
+    private lateinit var securityGuard: SecurityGuard
+
+    private val identityCreationStateDaoAsync = AppDatabase.getAppDatabase().identityCreationStateDaoAsync()
+    private val blockchainIdentityDataDaoAsync = AppDatabase.getAppDatabase().blockchainIdentityDataDaoAsync()
+
+    private val createIdentityNotification by lazy { CreateIdentityNotification(this) }
+
+    private val serviceJob = Job()
+    private val serviceScope = CoroutineScope(serviceJob + Dispatchers.Main)
+
+    lateinit var identityCreationState: IdentityCreationState
+    lateinit var blockchainIdentityData: BlockchainIdentityData
+
+    override fun onCreate() {
+        super.onCreate()
+        try {
+            securityGuard = SecurityGuard()
+        } catch (e: Exception) {
+            log.error("Unable to instantiate SecurityGuard", e)
+            stopSelf()
+            return
+        }
+        createIdentityNotification.startServiceForeground()
+    }
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        super.onStartCommand(intent, flags, startId)
+
+        if (intent != null) {
+
+            when (intent.action) {
+                ACTION_CREATE_IDENTITY -> handleCreateIdentityAction(intent)
+            }
+        }
+
+        return START_NOT_STICKY
+    }
+
+    private fun handleCreateIdentityAction(intent: Intent) {
+        val username = intent.getStringExtra(EXTRA_USERNAME)
+
+        val exceptionHandler = CoroutineExceptionHandler { _, exception ->
+            log.error("[${identityCreationState.state}(error)]", exception)
+            GlobalScope.launch {
+                updateState(identityCreationState.state, true)
+//                stopSelf()
```

Comment:
Changed to a TODO.  I don't know if we need to call stopSelf() when an exception is caught.  Does stopSelf() get called implicitly?  @tomasz-ludek 

https://github.com/dashevo/dash-wallet/pull/393/commits/17fca77c65ade7eb934521d2f0e9c35d93d04b22

Reply:
Let's not call the `stopSelf()` for now, it causes the notification informing about the error to be dismissed. It can be improved within https://dashpay.atlassian.net/browse/NMA-309.